File: dotnet/Sers/Sers.Core/Sers.Core.Station/Module/Message/ApiMessage.cs
using System;
using Sers.Core.Module.Message;
using System.Collections.Generic;
using Sers.Core.Module.Rpc;
using Sers.Core.Extensions;

namespace Sers.Core.Module.Api.Message
{
    public class ApiMessage: SersFile
    {
        public ApiMessage() { }

        public ApiMessage(ArraySegment<byte> oriData):base(oriData)
        {
        }

        public override List<ArraySegment<byte>> Files
        {
            get
            {
                var files = base.Files;
                if (files == null)
                {
                    base.Files= files = new List<ArraySegment<byte>>(2) { ArraySegmentByteExtensions.Null, ArraySegmentByteExtensions.Null };
                }
                return files;
            }
        }

        /// <summary>
        /// RpcContextData Files[0]
        /// </summary>
        public ArraySegment<byte> rpcContextData_OriData
        {
            get
            {
                return Files[0];
            }
            set
            {
                Files[0] = value;
            }
        }

        public void RpcContextData_OriData_Set(IRpcContextData rpcData)
        {
            rpcContextData_OriData = rpcData.PackageOriData();
        }


        /// <summary>
        /// value Files[1]
        /// 若为Request 则为 ArgValue
        /// 若为Reply则 为 ReturnValue
        /// </summary>
        public ArraySegment<byte> value_OriData
        {
            get
            {
                return Files[1];
            }
            set
            {
                Files[1] = value;
            }
        }

    }
}

File: java/Sers.ServiceStation/src/Sers/Core/Module/Message/ApiMessage.java
package Sers.Core.Module.Message;

import java.util.ArrayList;
import java.util.List;

import Sers.Core.Module.Api.Data.ApiReturn;
import Sers.Core.Module.Api.Data.ApiReturnBase;
import Sers.Core.Module.Rpc.RpcContextData;
import Sers.Core.Module.Rpc.RpcFactory;
import Sers.Core.Module.Serialization.Serialization;
import Sers.Core.Util.Data.ArraySegment;

public class ApiMessage extends SersFile {

	public ApiMessage() {
	}

	public ApiMessage(ArraySegment oriData) {
		super(oriData);
	}

	@Override
	public List<ArraySegment> Files_Get() {

		if (Files == null) {
			Files = new ArrayList<ArraySegment>();
			Files.add(null);
			Files.add(null);
		}
		return Files;
	}

	
	public ArraySegment rpcContextData_OriData_Get() {
		return Files_Get().get(0);
	}
	
	public void rpcContextData_OriData_Set(ArraySegment value) {
		Files_Get().set(0,value);
	}
	
	
	
	public ArraySegment value_OriData_Get() {
		return Files_Get().get(1);
	}
	
	public void value_OriData_Set(ArraySegment value) {
		Files_Get().set(1,value);
	}
	
	
	
	//region 扩展 InitByApiReturn InitAsApiRequestMessage
	  public ApiMessage InitByApiReturn(ApiReturnBase ret)
      {          
		  byte[] bytes=Serialization.Instance.serializeToBytes(ret);
          value_OriData_Set(new ArraySegment(bytes));
          return this;
      }
	
	  

      public ApiMessage InitAsApiRequestMessage(String route, Object arg)
      {
    	  byte[] argBytes=Serialization.Instance.serializeToBytes(arg);
    	  
    	  value_OriData_Set(new ArraySegment(argBytes)); 

          RpcContextData rpcData = RpcFactory.Instance.CreateRpcContextData.create().InitFromRpcContext();

       
          rpcData.route_Set(route);
          rpcData.http_url_Set(route);
          
          rpcContextData_OriData_Set(rpcData.PackageOriData());

          return this;
      }
	//endregion
	 
	
}
